### PR TITLE
Feature/stack interface

### PR DESCRIPTION
**Pull request by Dmytro-bruno**
1️⃣ feat: створено інтерфейс Stackable
Створено файл Stackable.java у пакеті lesson30stack

Всі методи описано з коментарями (JavaDoc)

Використано <E> generic для універсальності типів

2️⃣ feat: реалізовано клас Stack
Внутрішня реалізація на базі масиву з фіксованим розміром (DEFAULT_CAPACITY = 10)

Логіка push, pop, peek, isFull, isEmpty

Збереження LIFO-поведінки

3️⃣ feat: додано обробку помилок
Повідомлення в System.out.println, якщо стек повний або порожній

Легке логування для студента/викладача

4️⃣ test: написано тести StackTest.java
Тестування всіх публічних методів:

push / peek

pop

isEmpty / isFull

Перевірка реакції на крайні випадки (empty, full)

Покриття edge-case: pop з пустого стеку, push до повного

5. ♻️ Оновлено GitHub Actions:

Видалено main.yml та settings.yml

Додано новий ci.yml з підтримкою:

Java 17 (Temurin)

Maven cache

Командою mvn clean verify

6. ✅ Виправлено тести:

Оновлені імпорти у ShapeAreaTest.java і ShapePerimeterTest.java

Виправлено шляхи до класів Circle, Rectangle, Square, Shape

7. 📁 Переміщено класи з src/main/lesson28geometry → src/main/java/lesson28geometry

Для коректної Maven-сумісності з src/main/java як коренем пакету
*********************************************************************************
🎯 Що таке StackConsoleApp?
Це невелика демо-програма або симулятор, щоб вручну тестувати стек.

Вона не потрібна для автоматичного тестування, бо для цього є StackTest.java.

Але вона дуже корисна, якщо:

хочеш побачити, як воно працює в реальному застосунку

хочеш здати завдання з "інтерактивним" прикладом

хочеш протестувати руками замість юніт-тестів

🧠 Приклад розширення меню:
5. Перевірити чи стек порожній
6. Перевірити чи стек повний
7. Вивести всі елементи стеку

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      - name: ⬇️ Checkout repository
+        uses: actions/checkout@v3
+
+      - name: ☕ Set up JDK 17 (Temurin)
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: maven
+
```
Comment (nikmikhailov13) [inline .github/workflows/ci.yml:21]: а чому не java21

File: src/main/java/lesson30stack/Stack.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
+package lesson30stack;
+
+/**
+ * Клас Stack — реалізація структури даних стек (LIFO)
+ * @param <E> тип елементів у стеку
+ */
+public class Stack<E> implements Stackable<E> {
```
Comment (nikmikhailov13) [inline src/main/java/lesson30stack/Stack.java:7]: навіть Generics додали, круто

**Review by nikmikhailov13 — APPROVED** [review]
Стільки емоджі додали, прикольно 😁

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Dmytro-bruno) [reply .github/workflows/ci.yml:21]: Мені GPT порадив але потрібно це обговорити. Дякую

File: src/main/java/lesson30stack/Stack.java (hunk 2 above)
Comment (Dmytro-bruno) [reply src/main/java/lesson30stack/Stack.java:7]: А що можна було по іншому?)
